src/twisted/application/internet.py

```diff
@@ -97,7 +97,7 @@ class _AbstractServer(_VolatileDataService):
     """
 
     volatile = ['_port']
-    method = None
+    method = None  # type: Optional[str]
```

Comment:
I don't think `method` is actually optional: every concrete class provides a string. In Python 3.6 you could simply write `method: str` without assigning a value, but I don't know about any elegant 3.5 equivalent. Casting `None` to `str` would work and even though it looks ugly, it would simplify code that wants to use the `method` attribute.

Reply:
`method: str` doesn't work in Python 3.5

Reply:
I think we can define method as the empty string.